leshan-server-demo/src/main/java/org/eclipse/leshan/server/demo/LeshanServerDemo.java

```diff
@@ -299,24 +310,40 @@ public static void main(String[] args) throws FileNotFoundException {
         List<Certificate> trustStore = null;
         if (cl.hasOption("truststore")) {
             trustStore = new ArrayList<>();
-            File input = new File(cl.getOptionValue("truststore"));
 
-            // check input exists
-            if (!input.exists())
-                throw new FileNotFoundException(input.toString());
+            String trustStoreName = cl.getOptionValue("truststore");
 
-            // get input files.
-            File[] files;
-            if (input.isDirectory()) {
-                files = input.listFiles();
-            } else {
-                files = new File[] { input };
-            }
-            for (File file : files) {
+            if (trustStoreName.startsWith("file://")) {
+                // Treat argument as Java trust store
                 try {
-                    trustStore.add(SecurityUtil.certificate.readFromFile(file.getAbsolutePath()));
+                    Certificate[] trustedCertificates = SslContextUtil.loadTrustedCertificates(trustStoreName);
+                    trustStore.addAll(Arrays.asList(trustedCertificates));
                 } catch (Exception e) {
-                    LOG.warn("Unable to load X509 files {}:{} ", file.getAbsolutePath(), e.getMessage());
+                    System.err.println("Unable to load trust store : " + e.getMessage());
+                    e.printStackTrace();
+                    formatter.printHelp(USAGE, options);
```

Comment:
missing `return;` ?

Reply:
Will add -- it was kinda following the logic in CA file/path handling just give warnings and continue.

Reply:
added